### PR TITLE
Replace our custom search with Algolia search

### LINKED ISSUE #132: The "Report a problem or make a suggestion" link is not centered
image

--------------------------------

Please don't change anything below this point.

--------------------------------

 - Build ID: 2015-11-23T22:06:58+0000:d140164987ca743497f6d1295941525daa65a12a:cd53d644d5ba6bd89af5e778425fea90fac2092b
 - Page requested: /hhvm/
 - Page requested at: Tue, 24 Nov 2015 01:14:57 +0000

### LINKED ISSUE #312: Feature request: add as-you-type features to search
No drop down or interactive results

image

--------------------------------

Please don't change anything below this point.

--------------------------------

 - Build ID: 2016-03-25T19:44:16+0000:59945858b91b58253f39c91d5ff11f685e228917:ded7f690201f02c37531b00bb6a6fc0091e9945e
 - Page requested: /search
 - Page requested at: Tue, 26 Apr 2016 05:20:33 +0000
 - Controller: SearchController

### PR DESCRIPTION
**Pull request by JoelMarcey**
Fixes #312

screen shot 2016-05-11 at 10 15 16 am

### CODE REVIEW

**Comment by JoelMarcey** [comment]
@vjeux -- if you want to take a look at the CSS. I tried hacking something up relatively reasonable based on what we did nuclide.io, etc., but maybe there are better ways.

File: src/site/WebPageController.php [hunk 1]
```diff
@@ -357,4 +359,25 @@ function() {
...
   }
+
+  private function getAlgoliaDocSearchScript(): :script {
+    return <script type="text/javascript" src="https://cdn.jsdelivr.net/docsearch.js/1/docsearch.min.js"></script>;
+  }
+
+  private function getCallToAlgoliaDocSearchScript(): :script {
+    $code = <<<EOF
+// For Algolia search
+(function() {
```
Comment (vjeux) [inline src/site/WebPageController.php]: you don't need that function wrapper

File: src/site/WebPageController.php [hunk 2]
```diff
@@ -357,4 +359,25 @@ function() {
...
+
+  // Algolia
```
Comment (vjeux) [inline src/site/WebPageController.php]: you have two comments saying algolia, you can probably remove one of them

**Comment by vjeux** [comment]
Given the screenshot, it looks like there's some weird spacing going on. It is a lot wider than the other websites. Would be nice to get it fixed, otherwise looks good

**Pull request merged**